Docs: Table visualization update (#94495)
grafana/grafana: docs/sources/panels-visualizations/visualizations/table/index.md

# Table

Tables are a highly flexible visualization designed to display data in columns and rows.

> The table visualization can even take multiple datasets and provide the option to switch between them.

Reviewer: ```suggestion
The table visualization can take multiple datasets and provide the option to switch between them.
```

# Table

Tables are a highly flexible visualization designed to display data in columns and rows.

> They support various data types, including tables, time series, annotations, and raw JSON data.

Reviewer: Also this is a bit odd - what is a table(s) data type? or time series? 🤔

Author: So this is what it said back in v10.3 before our last round of content updates. Does this make more sense?

> Tables are very flexible, supporting multiple modes for time series and for tables, annotation, and raw JSON data.

Reviewer: I'm not convinced 😅 @leeoniya, any ideas?

Author: I mean time-series data is a kind of data. I'll fight you on that, lol. But I can take out "tables". I feel like by saying "column and rows" we kind of cover that.

Reviewer: yeah, I think that's fine. I don't have super strong opinions, just the wording around data types feels a bit odd. I think it's ok either way.

Author: Actually I think since that sentence isn't clear, we should just take it out. What do you think?
```suggestion
```

Reviewer 2: i'm not going to attempt wordsmithing here, but my general observation is that Table can render [as rows and columns] any and all data that our core panels, datasource plugins, and app plugins support ([DataFrames](https://grafana.com/developers/plugin-tools/key-concepts/data-frames)).

i dont know if there's value in trying to itemize a subset of DataFrame shapes when the real answer is that they can render "everything".

Author: Thanks for weighing in @leeoniya. I'm just gonna take that sentence out.

If you need to hide columns, you can do so using [data transformations](ref:data-transformation), [field overrides](#field-overrides), or by [building a query](ref:build-query) that returns only the needed columns.

> ## Debugging in tables

Reviewer: I feel like this section is from another story.

Author: Do you mean you don't think it fits here? If so, is there somewhere else you'd suggest putting it? We wanted to highlight that this is a thing you can do, so we wanted to make it prominent on this page.

Reviewer: I'm not sure where's a good place for it - but to me this is not related to table the visualization as you don't have any way to access options, customize it. I also feel like debugging is a big word for this. You're just visualizing your data in a table. But also it's been like this for a while and nobody said anything..  😅

Author: Okay, I'm gonna take it out. Maybe it might be useful to document this in other visualizations where you can switch to table view.

| [JSON View](#json-view)                   | Shows values formatted as code.                                                                                                                                                                                                                                                                                                               |
| [Image](#image)                           | If the field value is an image URL or a base64 encoded image, the table displays the image.                                                                                                                                                                                                                                                   |

##### Sparkline

This cell type shows values rendered as a sparkline.

> You can show sparklines using the [Time series to table transformation](ref:time-series-to-table-transformation) on data with multiple time series to process it into a format the table can show.

Author: ```suggestion
To show sparklines on data with multiple time series, use the [Time series to table transformation](ref:time-series-to-table-transformation) to process it into a format the table can show.
```

| [JSON View](#json-view)                   | Shows values formatted as code.                                                                                                                                                                                                                                                                                                               |
| [Image](#image)                           | If the field value is an image URL or a base64 encoded image, the table displays the image.                                                                                                                                                                                                                                                   |

##### Sparkline

This cell type shows values rendered as a sparkline.

> You can show sparklines using the [Time series to table transformation](ref:time-series-to-table-transformation) on data with multiple time series to process it into a format the table can show.

Author: Do you have to use this transformation to make this cell type work at all with time series data?

Reviewer: yes..

Author: I'm going to update this to make the language a bit stronger then.

![Table using sparkline cell type](/media/docs/grafana/panels-visualizations/screenshot-table-as-sparkline-v11.3.png)

> <!-- do you have to use this transformation to make this cell type work at all with time series data? -->

Author: ```suggestion
```
